application/models/Question.php

```diff
@@ -1725,16 +1723,30 @@ public function getScaledSubquestions()
      */
     public function questionThemeNameValidator()
     {
-        // As long as there is a question theme name, and it's not 'core', it's ok.
+        /* need a type */
+        if (empty($this->type)) {
+            return null;
+        }
+        /* not needed in child question */
+        if (!empty($this->parent_qid)) {
+            return null;
+        }
         if (!empty($this->question_theme_name) && $this->question_theme_name != 'core') {
-            return;
+            $criteria = new CDbCriteria();
+            $criteria->addCondition('question_type = :question_type AND name = :name');
+            $criteria->params = [':question_type' => $this->type, ':name' => $this->question_theme_name];
+            $questionTheme = QuestionTheme::model()->query($criteria, false);
+            if ($questionTheme) {
```

Comment:
What should happen if expected question theme doesn't exist?

Reply:
Reset to the QuestionTheme from question type.

If Question type didn't have theme : no solution. You can not update question->type

Reply:
Would like to call @olleharstedt or someone else to review this.
Another set of eyes would be good.

Reply:
If question type are disabled : we really have no way to use it (if survey is activated).